pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -196,6 +215,24 @@ func createDynamicMetadataMatcher(k, v string) *metadata.MetadataMatcher {
 		keys = []string{k}
 	}
 
+	var listMatcher *metadata.ListMatcher
+	if k == attrRequestGroups {
+		listMatcher = &metadata.ListMatcher{
```

Comment:
Why don't you have all these inside generateMetadataListMatcher?

Reply:
+1